fixit/rules/explicit_frozen_dataclass.py

```diff
@@ -0,0 +1,262 @@
+            @dc
+            class Cls: pass
+            """,
+            line=2,
+            expected_replacement="""
+            from dataclasses import dataclass as dc
+            @dc(frozen=True)
+            class Cls: pass
+            """,
+        ),
+        Invalid(
+            """
+            from dataclasses import dataclass as dc
+            @dc()
+            class Cls: pass
+            """,
+            line=2,
+            expected_replacement="""
+            from dataclasses import dataclass as dc
+            @dc(frozen=True)
+            class Cls: pass
+            """,
+        ),
+        Invalid(
+            """
+            from dataclasses import dataclass as dc
+            @dc(other_kwarg=False)
+            class Cls: pass
+            """,
+            line=2,
+            expected_replacement="""
+            from dataclasses import dataclass as dc
+            @dc(other_kwarg=False, frozen=True)
+            class Cls: pass
+            """,
+        ),
+        Invalid(
+            """
+            import dataclasses as dc
+            @dc.dataclass
+            class Cls: pass
+            """,
+            line=2,
+            expected_replacement="""
+            import dataclasses as dc
+            @dc.dataclass(frozen=True)
+            class Cls: pass
+            """,
+        ),
+        Invalid(
+            """
+            import dataclasses as dc
+            @dc.dataclass()
+            class Cls: pass
+            """,
+            line=2,
+            expected_replacement="""
+            import dataclasses as dc
+            @dc.dataclass(frozen=True)
+            class Cls: pass
+            """,
+        ),
+        Invalid(
+            """
+            import dataclasses as dc
+            @dc.dataclass(other_kwarg=False)
+            class Cls: pass
+            """,
+            line=2,
+            expected_replacement="""
+            import dataclasses as dc
+            @dc.dataclass(other_kwarg=False, frozen=True)
+            class Cls: pass
+            """,
+        ),
+    ]
+
+    def visit_ClassDef(self, node: cst.ClassDef) -> None:
+        for d in node.decorators:
+            decorator = d.decorator
+            if QualifiedNameProvider.has_name(
+                self,
+                decorator,
+                QualifiedName(
+                    name="dataclasses.dataclass", source=QualifiedNameSource.IMPORT
+                ),
+            ):
+                if isinstance(decorator, cst.Call):
+                    func = decorator.func
+                    args = decorator.args
+                else:  # decorator is either cst.Name or cst.Attribute
+                    args = ()
+                    func = decorator
+
+                # pyre-fixme[29]: `typing.Union[typing.Callable(tuple.__iter__)[[], typing.Iterator[Variable[_T_co](covariant)]], typing.Callable(typing.Sequence.__iter__)[[], typing.Iterator[cst._nodes.expression.Arg]]]` is not a function.
```

Comment:
Not sure if this is a proper/valid way to deal with the pyre error. But it seems like there might be some upstream blockers with pyre based on what I found internally.

Reply:
Did pyre complain about the `any()` is not a function?

Reply:
Oh thanks for the catch!